app/flatpak-builtins-build-bundle.c

```diff
@@ -531,6 +523,50 @@ build_oci (OstreeRepo *repo, const char *collection_id, GFile *dir,
   return TRUE;
 }
 
+static gboolean
+_repo_resolve_rev (OstreeRepo *repo, const char *ref, char **out_rev,
+                   GCancellable *cancellable, GError **error)
+{
+  g_autoptr(GError) my_error = NULL;
+
+  g_return_val_if_fail (repo != NULL, FALSE);
+  g_return_val_if_fail (ref != NULL, FALSE);
+  g_return_val_if_fail (out_rev != NULL, FALSE);
+
+  if (ostree_repo_resolve_rev (repo, ref, FALSE, out_rev, &my_error))
+    return TRUE;
+  else
+    {
+      g_autoptr(GHashTable) collection_refs = NULL;  /* (element-type OstreeCollectionRef utf8) */
+
+      /* Fall back to iterating over every collection-ref. We can't use
+       * ostree_repo_resolve_collection_ref() since we don't know the
+       * collection ID. */
+      if (!ostree_repo_list_collection_refs (repo, NULL, &collection_refs,
+                                             OSTREE_REPO_LIST_REFS_EXT_NONE,
+                                             cancellable, error))
+        return FALSE;
+
+      /* Take the checksum of the first matching ref. There's no point in
+       * checking for duplicates because (a) it's not possible to install the
+       * same app from two collections in the same flatpak installation and (b)
+       * ostree_repo_resolve_rev() also takes the first matching ref. */
+      GLNX_HASH_TABLE_FOREACH_KV (collection_refs, const OstreeCollectionRef *, c_r,
+                                  const char*, checksum)
+        {
+          if (g_strcmp0 (c_r->ref_name, ref) == 0)
+            {
+              *out_rev = g_strdup (checksum);
+              return TRUE;
+            }
+        }
+
+      g_assert (*out_rev == NULL);
```

Comment:
Shouldn't this be a `g_return_val_if_fail` at the top of the function? It is not under this function's (direct) control.

Reply:
Yeah you're right. Pushed a fixup